cf-rd/src/main/java/org/eclipse/californium/tools/resources/RDNodeResource.java

```diff
@@ -197,7 +198,10 @@ public CoapResource addNodeResource(String path) {
 				}
 			}
 			if (!resourceExist) {
-				subResource = new RDTagResource(next, true, this);
+				if (scanner.hasNext())//if the resource is not the end resource
+					subResource = new RDTagResource(next, true, this);
+				else
+					subResource = addEndResource(next, attributes);
```

Comment:
what's the purpose of this if the `addEndResource()` method also adds an `RDTagResource`?